HW3_IzzyGaw.R

```diff
@@ -0,0 +1,115 @@
+	  approxy <- x/newguess
+	  newguess <- (newguess+approxy)/2 
+	  error = newguess - approxy
+	  print(error < tol) 
+	  print(newguess)
+	  print(error)
+	}
+	return( newguess )
+
+}
+
+## Add an argument for verbose=FALSE in the function definition
+## Change the print lines to be conditional 
+## statements to be run only if the user chooses verbose=TRUE
+## The function should then return only the final solution
+
+#if else # conditional statements, else is optional
+####EX. if x>y print(x) else print(y)
+#while #while the condition is true, run this code.
+####EX. while (tolerance > .001)
+#repeat #repeat. "break" is required to stop execution
+####EX. if (condition) {break}
+##########
+
+
+
+##########################
+#########MY WORK##########
+##########################
+
+mysqrt <- function( x=81, tol=.000001, myguess=27, verbose=FALSE){  #This sets up the function and its arguments. It establishes the x and myguess (which both are used to find the square root), the tolerance (which is the amount of deviation allowed from the actual answer), and verbose (which if turned off will not show all the output of the conditionals (default), but if TRUE will show). 
+
+	newguess <- myguess #renames myguess to newguess
+	newguess  #check
+	
+	if (x>0) { #this makes it so you cannot use negatives in the input, and if you do, you get a custom error message
+	approxy <- x/newguess #divides x by newguess and renames approxy
+	approxy #check
+	newguess <- (newguess+approxy)/2 #divides the sum of new guess and approxy by 2 and renaming to newguess.
+	newguess #check
+	error = newguess - approxy #finding the error 
+	error <- abs(error) #making sure the error is positive by using the absolute value function
+	error < tol #made the tolerance higher than the error
+	if (verbose==T) {print(error < tol)} #if the user wanted, this function would print if the error is larger or smaller than the tolerance.
+	if (verbose==T) {print(newguess)} #if the user wanted, this function would print the value of newguess.
+	if (verbose==T) {print(error)} #if the user wanted, this function would print the value of the error.
+
+	while ( error > tol ) { #this function says, if the error is larger than the tolerance, then ...
+	  approxy <- x/newguess  #divide x/newguess
+	  newguess <- (newguess+approxy)/2  #add newguess and approxy then divide by 2
+	  error = newguess - approxy #subtract approxy from newguess
+	  error <- abs(error)  #find the absolute value of error and rename to "error"
+	  if (verbose==T) {print(error < tol)} #if the user wanted, this function would print if the error is larger or smaller than the tolerance.
+	  if (verbose==T) {print(newguess)} #if the user wanted, this function would print the value of newguess.
+	  if (verbose==T) {print(error)} #if the user wanted, this function would print the value of the error.
+	}
+	return( newguess ) # 
+	}else print("BAD. DO NOT USE NEGATIVES") #this is the custom error message
+
+}
```

Comment:
Woah! This is way too cluttered. IF you were to look at this function a year from now it would be very hard to understand. Only comment what you need to understand the function. THere is such a thing as clogging up your code with too many comments.